pkg/apiserver/endpoints/handlers/get.go

```diff
@@ -157,6 +157,10 @@ func getRequestOptions(req *http.Request, scope handlers.RequestScope, into runt
 			startingIndex = 3
 		}
 
+		if len(requestInfo.Parts) < 2 {
```

Comment:
This will break the following API paths:
```
/{object-type}/{object-name}/{metric-name...}
/{object-type}/*/{metric-name...}
```
You can find all of the supported API paths of the custom metrics API here: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md#api-paths

Something else to note is that the getRequestOptions function is taken from [upstream Kubernetes](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/get.go#L132-L167), so we most likely don't want to change it and might prefer acting on the caller by verifying that the arguments provided to the function are correct. But it may also be that the Kubernetes implementation doesn't need to handle this edge case, where the provided API path is incorrect.

Reply:
oh, Got it, thanks! I will try to find another place to do the validation.